Factory Pattern/src/OrderEnum.java

```diff
@@ -0,0 +1,14 @@
+public enum OrderEnum {
+    FOOD(new FoodOrder()),
+    CLOTHING(new ClothingOrder()),
+    ELECTRONICS(new ElectronicsOrder());
```

Comment:
the problem is, it is one object for each customer. suppose 2 persons orders Food. they will get same instance of `FoodOrder`

Reply:
Here its returning one single object for each class. We should get different Objects right ?